src/main/java/me/zeroeightsix/kami/module/modules/movement/AutoRemount.java

```diff
@@ -30,12 +31,20 @@
     private Setting<Boolean> pig = register(Settings.b("Pig", true));
     private Setting<Boolean> llama = register(Settings.b("Llama", true));
     private Setting<Float> range = register(Settings.floatBuilder("Range").withMinimum(1.0f).withValue(2.0f).withMaximum(10.0f).build());
+    private Setting<Integer> delay = register(Settings.integerBuilder("Delay").withMinimum(-1).withValue(5).withMaximum(100).build());
+
+    private int timer = delay.getValue();
 
     public void onUpdate() {
-        // we don't need to do anything if we're already riding.
         if (mc.player.isRiding())
             return;
 
+        // Bad. There is no way to check if a setting has been changed, this workaround works
```

Comment:
Uhhhh that's why we have the setting listener?

Reply:
Look at crystalaura for an example of how to use it

Reply:
Hey! You're right...

Except the setting listener is automatic... And you don't need to manually configure it...

In fact it runs every time a setting is accessed.

`    public SettingListeners settingListener;

    public interface SettingListeners {
        public void onSettingChange(final Setting setting);
    }

    public boolean setValue(T value) {
        T old = getValue();
        if (!restriction.test(value))
            return false;
        this.value = value;
        consumer.accept(old, value);
        if (settingListener != null)
            settingListener.onSettingChange(this);
        return true;
    }` 

I'd remove it from CrystalAura if I waere you, but whatever bro. I'll remove my comment :^)

Reply:
It has a good reason for being in crystalaura, it's the cleanest way to check if a setting was changed

Reply:
Did you just not read the code I just sent you? The setting listener already works implicitly. You don't need to add it again. I don't understand why the original writer made the settinglistener portion of the code public, he potentially set it there in case he'd needed to override it in some way, but you don't really need to add it anywhere if it's just to check if a setting was changed.

Reply:
I don't really understand

You're saying it should be changed?

Reply:
Read zeroeightysix' original "Setting" code. It already does the job of checking when a value was updated. I blame the lack of implicit gets and sets for the misunderstanding. I think the reason why zeroeightysix made the SettingListeners object public was to override it for explicit polling if that was needed/

Either ways, my original comment was misinformation and I'll remove it when I have time to.

Reply:
I wrote the listeners, and that was probably me misunderstanding how the original setting system works. If you can explain more clearly how I should fix it, I'd be happy to

Reply:
The entire listener system looks, from my perspective at least, to be unnecessary. If you wish to keep them, I'd recommend making an event that's called whenever a setting is changed to retrigger value assignments.

Reply:
It's used to do other things, such as reset modules to default settings. It's essentially a lighter listener